### DISCUSSION TITLE
Question about Wazuh HA deployment

### DISCUSSION THREAD

**Discussion by jackylam2** [wazuh/wazuh#14192] [Q&A] [answered]
Hi,

I installed lastest Wazuh (4.3) with multi-node cluster and it works well. 

However, my intend thought is to deploy a HA wazuh (which mean I can always access or manage alert or configuration by web-interface)

My setup is 2 wazuh indexers , 2 wazuh server (1 master and 1 worker), 1 wazuh dashboard and 1 nginx as load balancer. Nginx load balancer is for wazuh cluster (1 master and 1 worker).

Below are the testing I tried:
1. Stopped one wazuh indexer -> Wazuh web-interface failed 
![image](https://user-images.githubusercontent.com/81209411/178140039-08c3a25d-8b37-4c29-831c-c9d802a2fe4d.png)
2. Stopped one wazuh server (master) -> Wazuh web-interface failed with API connection failed; Agents auto switched to worker.
![image](https://user-images.githubusercontent.com/81209411/178140148-3e401b12-230a-4379-8ea3-2288352592c8.png)
3. Stopped one wazuh server (worker) ->  Wazuh web-interface normal; Agents auto switched to master.

According the test results, I cannot lose one of the indexer or wazuh server (master). I learned wazuh dashboard is connected to wazuh server (master) so wazuh web-interface will fail when it is down by reading wazuh doc. Also, it seems the wazuh cluster HA is on backend (agents will auto switch to other wazuh server)

Therefore, am I set the HA right? (e.g. do I need 3 three nodes in the cluster)

**Comment by mdiego92** [comment, marked as answer]
Hello jackylam2,

The master is essential to perform the tasks mentioned [here](https://documentation.wazuh.com/current/development/wazuh-cluster.html#types-of-nodes). Therefore, and regarding your question, if the master node is the one going down, all cluster's related tools won't be available, which include:

- Agent registration.
- Agent deletion.
- Rules, decoders, and CDB list synchronization (with other nodes).
- Configuration of agents grouping.

It is important to remark that the Wazuh API is located on the master node, so if it goes down, you won't be able to use the Wazuh UI until the master is available again.

However, even though those tools will not work, the running worker nodes will still be able to carry out their manager's tasks:

- Getting logs from the agents that report to them.
- Matching logs with decoders and rules, and sending any alert generated to Wazuh Indexer via Filebeat.

Therefore, although you cannot use the Wazuh Dashboard plugin, you could still check all the alerts generated by the worker in Wazuh Dashboard.

If you use a load balancer or failover mode, the agents which were reporting to the master node will be distributed among the nodes still available, so corresponding alerts will still be shown in Wazuh Dashboard and nothing will be lost.

Please let us know if this could solve your request or if you have any remaining questions.

Thank you very much.

**Reply by jackylam2** [reply]
Hi mdiego92,

Got it, but may I ask why  Wazuh web-interface failed when I stopped one indexer?

Thank you

**Reply by mdiego92** [reply]
Hello jackylam2,

If Wazuh Indexer is stopped you wouldn't be able to log in to the web interface since it is the service that stores and uses the `/usr/share/wazuh-indexer/plugins/opensearch-security/securityconfig/internal_users.yml` file to access to Wazuh-Dashboard.

Please refer to the following documentation:
https://documentation.wazuh.com/current/user-manual/securing-wazuh/wazuh-indexer.html

Please let us know if you have any remaining questions.

Thank you very much.

**Reply by jackylam2** [reply]
Hi mdiego92,

Got it, but I think I had setup a indexer cluster (2x wazuh indexer) and they should be in sync (HA)? Then, when I failed one indexer, the Wazuh-Dashoboard will auto to connect to other indexer?

Thank you.

**Reply by mdiego92** [reply]
Hello jackylam2,

The best practice for an Indexer cluster would be using three nodes, as if you have one node failing, you would still have the cluster running. If you have a two-node cluster, it will create a split-brain if the network gets lost and both nodes are up.

As long as you have the cluster up and running properly, Wazuh Dashboard will be able to connect to Wazuh Indexer without issues.
Another point to consider is data indexing. It's important that it keeps running with the cluster working, even if one node is failing. This is also related to how many shards and replicas you configured within your cluster.

**Please refer to the following article for more information**: https://www.elastic.co/blog/how-many-shards-should-i-have-in-my-elasticsearch-cluster (Since V4.3 Wazuh Indexer works with OpenSearch, a fork of the Elasticsearch service)

Please let us know if you have any questions or doubts and we will happily answer back!

Thank you very much.

**Reply by jackylam2** [reply]
Hi mdiego92,

Got it, Thank you so much for your help

**Reply by SaeidSec** [reply]
If master node stop dashboard fail so how to solve this dashboard fail issues if master node is down? Is there any solution?  🙂

**Comment by SaeidSec** [comment]
If master node stop dashboard fail so how to solve this dashboard fail issues if master node is down? Is there any solution?  🙂

**Comment by SaeidSec** [comment]
If master node stop dashboard fail so how to solve this dashboard fail issues if master node is down? Is there any solution?  🙂
